synfig-studio/src/gui/workarea.cpp

```diff
@@ -629,41 +615,34 @@ WorkArea::load_meta_data()
 	if(data.size() && (data=="0" || data[0]=='f' || data[0]=='F'))
 		set_background_rendering(false);
 
-	data=canvas->get_meta_data("guide_x");
-	get_guide_list_x().clear();
+	//for the guide to be stored we have to store 2 floats x,y + 1 float angle in rad + a true or false
+	data=canvas->get_meta_data("guide");
+	get_guide_list().clear();
 	while(!data.empty())
 	{
 		String::iterator iter(find(data.begin(),data.end(),' '));
-		String guide(data.begin(),iter);
-	    ChangeLocale change_locale(LC_NUMERIC, "C");
-
-		if(!guide.empty())
-			get_guide_list_x().push_back(stratof(guide));
-
-		if(iter==data.end())
-			data.clear();
-		else
-			data=String(iter+1,data.end());
-	}
-	//sort(get_guide_list_x());
-
-	data=canvas->get_meta_data("guide_y");
-	get_guide_list_y().clear();
-	while(!data.empty())
-	{
-		String::iterator iter(find(data.begin(),data.end(),' '));
-		String guide(data.begin(),iter);
-	    ChangeLocale change_locale(LC_NUMERIC, "C");
-
-		if(!guide.empty())
-			get_guide_list_y().push_back(stratof(guide));
+		String guide(data.begin(),iter);//this now contains the four things
+		std::vector<String> guide_components(4);
+		int i = 0;
+		for (auto character: guide){
+			if(character != '*')
+				guide_components.at(i).push_back(character);
+			else
+				++i;
+		}
+		if(!guide.empty()){
```

Comment:
I'm not sure i'm catching what you mean here. `guide_components` is declared to be of size 4, so it will always have four elements even if one of them is empty. Maybe you mean to check that if these elements are non empty themselves. But how would this be better ? As long as the guide string isn't empty it's guaranteed to contain the four elements which are always inserted when the metadata is saved/set.

Reply:
You should at least log it via `synfig::warning()`

Reply:
Change to `if (guide.size() != 3)`